rtgui/controlspotpanel.cc

```diff
@@ -1912,60 +1908,60 @@ void ControlSpotPanel::deleteControlSpot(const int id)
     disableParamlistener(false);
 }
 
-ControlSpotPanel::SpotEdited* ControlSpotPanel::getEditedStates()
+ControlSpotPanel::SpotEdited ControlSpotPanel::getEditedStates() const
 {
     // printf("getEditedStates\n");
 
-    SpotEdited* se = new SpotEdited();
+    SpotEdited se = {};
 
     if (nbSpotChanged_) {
-        se->nbspot = true;
+        se.nbspot = true;
         // nbSpotChanged_ = false;
     } else {
-        se->nbspot = false;
+        se.nbspot = false;
     }
 
     if (selSpotChanged_) {
-        se->selspot = true;
+        se.selspot = true;
         // selSpotChanged_ = false;
     } else {
-        se->selspot = false;
+        se.selspot = false;
     }
 
     if (nameChanged_) {
-        se->name = true;
+        se.name = true;
         // nameChanged_ = false;
     } else {
-        se->name = false;
+        se.name = false;
     }
 
     if (visibilityChanged_) {
-        se->isvisible = true;
+        se.isvisible = true;
         // visibilityChanged_ = false;
     } else {
-        se->isvisible = false;
-    }
-
-    se->shape = shape_->get_active_text() != M("GENERAL_UNCHANGED");
-    se->spotMethod = spotMethod_->get_active_text() != M("GENERAL_UNCHANGED");
-    se->sensiexclu = sensiexclu_->getEditedState();
-    se->structexclu = structexclu_->getEditedState();
-    se->struc = struc_->getEditedState();
-    se->shapeMethod = shapeMethod_->get_active_text() != M("GENERAL_UNCHANGED");
-    se->locX = locX_->getEditedState();
-    se->locXL = locXL_->getEditedState();
-    se->locY = locY_->getEditedState();
-    se->locYT = locYT_->getEditedState();
-    se->centerX = centerX_->getEditedState();
-    se-> centerY = centerY_->getEditedState();
-    se->circrad = circrad_->getEditedState();
-    se->qualityMethod = qualityMethod_->get_active_text() != M("GENERAL_UNCHANGED");
-    se->transit = transit_->getEditedState();
-    se->thresh = thresh_->getEditedState();
-    se->iter = iter_->getEditedState();
-    se->balan = balan_->getEditedState();
-    se->transitweak = transitweak_->getEditedState();
-    se->avoid = !avoid_->get_inconsistent();
+        se.isvisible = false;
+    }
+
```

Comment:
@Floessie Why not
```
    se.nbspot = nbSpotChanged_;
    se.selspot = selSpotChanged_;
    se.name = nameChanged_;
    se.isvisible = visibilityChanged_;
```
instead of the if/else?

Reply:
@heckflosse Of course. I was just comparing ASAN output with `new ...` lines and missed the obvious...

Reply:
BTW: `controlspotpanel.*` needs a full sweep. Maybe next week...